tests/integration/test_attachmentsv2.py

```diff
@@ -593,6 +600,75 @@ def test_span_attachment_ds_drop(mini_sentry, relay, rule_type):
     assert mini_sentry.captured_outcomes.empty()
 
 
+@pytest.mark.parametrize(
+    "rule_type",
+    ["project", "trace"],
+)
+@pytest.mark.parametrize(
+    "should_drop",
+    [False, True],
+)
+def test_trace_attachment_ds(mini_sentry, relay, rule_type, should_drop):
+    project_id = 42
+    project_config = mini_sentry.add_full_project_config(project_id)
+    project_config["config"]["features"] = [
+        "projects:trace-attachment-processing",
+    ]
+    sample_rate = 0.0 if should_drop else 1.0
+    add_sampling_config(project_config, sample_rate=sample_rate, rule_type=rule_type)
+
+    relay = relay(mini_sentry, options=TEST_CONFIG)
+
+    envelope = Envelope()
+    envelope.headers["trace"] = {
+        "trace_id": "5b8efff798038103d269b633813fc60c",
+        "public_key": project_config["publicKeys"][0]["publicKey"],
+    }
+
+    metadata = create_attachment_metadata()
+    body = b"span attachment content"
+    metadata_bytes = json.dumps(metadata, separators=(",", ":")).encode("utf-8")
+    combined_payload = metadata_bytes + body
+
+    if should_drop:
+        assert mini_sentry.get_outcomes(n=2) == [
+            {
+                "timestamp": time_within_delta(),
+                "org_id": 1,
+                "project_id": 42,
+                "key_id": 123,
+                "outcome": 1,
+                "reason": "Sampled:0",
+                "category": DataCategory.ATTACHMENT.value,
+                "quantity": len(body),
+            },
+            {
+                "timestamp": time_within_delta(),
+                "org_id": 1,
+                "project_id": 42,
+                "key_id": 123,
+                "outcome": 1,
+                "reason": "Sampled:0",
+                "category": DataCategory.ATTACHMENT_ITEM.value,
+                "quantity": 1,
+            },
+        ]
+    envelope.add_item(
+        Item(
+            payload=PayloadRef(bytes=combined_payload),
+            type="attachment",
+            headers={
+                "content_type": "application/vnd.sentry.attachment.v2",
+                "meta_length": len(metadata_bytes),
+                "length": len(combined_payload),
+            },
+        )
+    )
```

Comment:
I think this go messed up during some merge, order should be swapped and envelope send?